src/AutoRest.CSharp/Mgmt/Decorator/CodeModelExtension.cs

```diff
@@ -0,0 +1,169 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using System.Linq;
+using System.Text;
+using System.Text.RegularExpressions;
+using AutoRest.CSharp.AutoRest.Plugins;
+using AutoRest.CSharp.Input;
+using AutoRest.CSharp.Utilities;
+
+namespace AutoRest.CSharp.Mgmt.Decorator
+{
+    internal static class CodeModelExtension
+    {
+        private static readonly ConcurrentDictionary<string, string> _cache = new ConcurrentDictionary<string, string>();
+
+        public static void UpdateFrameworkTypes(this IEnumerable<Schema> allSchemas)
+        {
+            foreach (var schema in allSchemas)
+            {
+                if (schema is not ObjectSchema objSchema)
+                    continue;
+
+                foreach (var property in objSchema.Properties)
+                {
+                    if (property.Language.Default.Name.EndsWith("Uri"))
+                        property.Schema.Type = AllSchemaTypes.Uri;
+                }
+            }
+        }
+
+        public static void UpdateSubscriptionIdForAllResource(this CodeModel codeModel)
+        {
+            bool setSubParam = false;
+            foreach (var operationGroup in codeModel.OperationGroups)
+            {
+                foreach (var op in operationGroup.Operations)
+                {
+                    foreach (var p in op.Parameters)
+                    {
+                        //updater the first subscriptionId to be 'method'
+                        if (!setSubParam && p.Language.Default.Name.Equals("subscriptionId", StringComparison.OrdinalIgnoreCase))
+                        {
+                            setSubParam = true;
+                            p.Implementation = ImplementationLocation.Method;
+                        }
+                        //updater the first subscriptionId to be 'method'
+                        if (p.Language.Default.Name.Equals("apiVersion", StringComparison.OrdinalIgnoreCase))
```

Comment:
why ignore case?

Reply:
Well, this is not introduced by my PR, I just move this function from MgmtOutputLibrary to here, since this function also works as an extension method of codeModel.
@m-nash do you know the answer?